### PR TITLE
Drop Rails 4.2 support

### LINKED ISSUE #202: Drop Rails 4.2 support
Rails 4.2 is now EOL.  We will be dropping support for it in v0.16.0.

If you have concerns about this, please comment on this issue.

### PR DESCRIPTION
**Pull request by olbrich**
- Drops conditional handling of Rails versions < 5.0

Fixes #202

### CODE REVIEW

File: .github/workflows/test_suite.yml
Comment (olbrich) [inline .github/workflows/test_suite.yml:15]: Dropping 4.2 here and from conditionals is the major change here

File: .github/workflows/test_suite.yml
Comment (oniofchaos) [inline .github/workflows/test_suite.yml:28]: Thanks for the format update here. Much easier to read. :)

File: Gemfile
Comment (oniofchaos) [inline Gemfile:19]: Can update this comment to reflect that we _just_ need to update the tests now.

File: README.md
Comment (oniofchaos) [inline README.md:324]:
```suggestion
- Rails 5 onwards (<http://github.com/rails/rails>)
```

File: test/support/integration_helpers.rb
Comment (oniofchaos) [inline test/support/integration_helpers.rb:8]:
```suggestion
```

**Review by oniofchaos — COMMENTED** [review]
LOVE IT! 

One small thing I think we can clean up:

Since we aren't using Mongoid.4 anymore (since that was for Rails 4.2), we can get rid of the conditionals in these two files.

https://github.com/devise-security/devise-security/blob/f2d3fe8a4a2a28bc6764b6c3b437e3887cb4ebe0/test/dummy/config/mongoid.yml
https://github.com/devise-security/devise-security/blob/f2d3fe8a4a2a28bc6764b6c3b437e3887cb4ebe0/test/support/mongoid.yml

File: test/support/integration_helpers.rb
Comment (olbrich) [inline test/support/integration_helpers.rb:22]:
```suggestion
```

**Review by oniofchaos — APPROVED** [review]
LGTM, not sure if you want to coordinate anything else before merging so I'll let you handle that.

File: Gemfile
Comment (tmr08c) [inline Gemfile:15]: Curious about why we need to specify a version here

File: test/dummy/config/mongoid.yml
Comment (tmr08c) [inline test/dummy/config/mongoid.yml:2]: Is this why we specify `mongoid` to be version `6` in our base `Gemfil`?

**Review by tmr08c — COMMENTED** [review]
Late review, but 👍

**Pull request merged**
